### ISSUE TITLE
Topology page loads white screen in webConsole

### ISSUE THREAD

**Issue by klzsysy** [openshift/console#8499]
## Describe the bug
In webConsole view of the developers ，topology page load white screen，all the namespace is the same problem，but Other page is normal

![image](https://user-images.githubusercontent.com/3226923/112821532-64d59300-90b9-11eb-975d-8e1bc505631c.png)

![image](https://user-images.githubusercontent.com/3226923/112821577-728b1880-90b9-11eb-81c9-820615bdfd7c.png)

Version
 
4.7.0-0.okd-2021-03-28-152009

How reproducible

100%

It is reproduceable in 4.7, but not in 4.6

**Comment by jessedye** [comment]
Also seeing this bug in 4.7 as well.

**Comment by evertmulder** [comment]
Just upgraded to 4.7.0-0.okd-2021-04-11-124433. The issue still exists.

**Comment by spadgett** [comment]
@christianvogt @andrewballantyne

**Comment by invincibleJai** [comment]
we did try to verify this on `4.7.0-0.okd-2021-03-28-152009` and we couldn't reproduce it, PFA gif. 

![Apr-19-topology](https://user-images.githubusercontent.com/5129024/115193165-6c60e880-a109-11eb-8d9e-be08c101288a.gif)

Looks like we need, some more information to reproduce it. cc @karthikjeeyar

**Comment by evertmulder** [comment]
@invincibleJai Hi I uploaded a screen recording showing the disappearing screen: https://youtu.be/4iqTSQbT2wQ
I am using 4.7.0-0.okd-2021-04-11-124433, IPI install on VMWare. I have this issue on all 4.7 releases. 
Please let me know if I can provide more info.

**Comment by invincibleJai** [comment]
> @invincibleJai Hi I uploaded a screen recording showing the disappearing screen: https://youtu.be/4iqTSQbT2wQ
> I am using 4.7.0-0.okd-2021-04-11-124433, IPI install on VMWare. I have this issue on all 4.7 releases.
> Please let me know if I can provide more info.

Thanks @evertmulder, for providing us with details. Based on the above comments, is it happening on all namespaces?
Looks like we need, some more information to reproduce it

- What was the user role?
- Was there any Operator installed on the cluster?

**Comment by evertmulder** [comment]
This is happening in all namespaces (with pods)
My role is cluster admin, also happens as kube-admin user.
I do not have additional cluster operators installed.
I am using OVNKubernetes and configured to use a proxy in the openshift-installer.

```bash
oc get co
NAME                                       VERSION                         AVAILABLE   PROGRESSING   DEGRADED   SINCE
authentication                             4.7.0-0.okd-2021-04-11-124433   True        False         False      5d19h
baremetal                                  4.7.0-0.okd-2021-04-11-124433   True        False         False      31d
cloud-credential                           4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
cluster-autoscaler                         4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
config-operator                            4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
console                                    4.7.0-0.okd-2021-04-11-124433   True        False         False      5h1m
csi-snapshot-controller                    4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
dns                                        4.7.0-0.okd-2021-04-11-124433   True        False         False      21d
etcd                                       4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
image-registry                             4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
ingress                                    4.7.0-0.okd-2021-04-11-124433   True        False         False      55d
insights                                   4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
kube-apiserver                             4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
kube-controller-manager                    4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
kube-scheduler                             4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
kube-storage-version-migrator              4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
machine-api                                4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
machine-approver                           4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
machine-config                             4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
marketplace                                4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
monitoring                                 4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
network                                    4.7.0-0.okd-2021-04-11-124433   True        False         False      31d
node-tuning                                4.7.0-0.okd-2021-04-11-124433   True        False         False      6d13h
openshift-apiserver                        4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
openshift-controller-manager               4.7.0-0.okd-2021-04-11-124433   True        False         False      6d13h
openshift-samples                          4.7.0-0.okd-2021-04-11-124433   True        False         False      6d13h
operator-lifecycle-manager                 4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
operator-lifecycle-manager-catalog         4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
operator-lifecycle-manager-packageserver   4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
service-ca                                 4.7.0-0.okd-2021-04-11-124433   True        False         False      119d
storage                                    4.7.0-0.okd-2021-04-11-124433   True        False         False      6d12h
```

**Comment by christianvogt** [comment]
@evertmulder could you tell us how many Templates you have in the `openshift` namespace
`oc get templates -n openshift --output json | jq '.items' | jq length`

Could you go to the dev catalog and see if you encounter the same issue.

@invincibleJai i suspect the issue is this line here:
https://github.com/openshift/console/blob/39b08772b0fcce105c81319b796d897e358ece95/frontend/packages/dev-console/src/components/catalog/providers/useTemplates.tsx#L87

should be:
```
setTemplates(metadata ?? []);
```

**Comment by klzsysy** [comment]
> @evertmulder could you tell us how many Templates you have in the `openshift` namespace
> `oc get templates -n openshift --output json | jq '.items' | jq length`
> 
> Could you go to the dev catalog and see if you encounter the same issue.
> 
> @invincibleJai i suspect the issue is this line here:
> https://github.com/openshift/console/blob/39b08772b0fcce105c81319b796d897e358ece95/frontend/packages/dev-console/src/components/catalog/providers/useTemplates.tsx#L87
> 
> should be:
> 
> ```
> setTemplates(metadata ?? []);
> ```

```sh
❯ oc get templates -n openshift --output json | jq '.items' | jq length
0
```

**Comment by evertmulder** [comment]
@christianvogt @invincibleJai  I think you guy are absolutely right. 

```
 oc get templates -n openshift --output json | jq '.items' | jq length
0
```

I also created a template in the `openshift` namespace and things are working again without a problem!

**Comment by christianvogt** [comment]
OpenShift Samples Operator:
> During installation, the Operator creates the default configuration object for itself and then creates the sample imagestreams and templates, including quickstart templates.

So disabling the samples operator caused the issue.

We will definitely look at fixing this issue.

**Comment by spadgett** [comment]
Fixed by #8732
